### PR TITLE
[tools] expand helm renderer

### PR DESCRIPTION
**Pull request by Dmitrykob**
## Description
 
Make generic solution for rendering helm templates.

## Why do we need it, and what problem does it solve?

Easily expandable tool that should help render helm templates. Issue https://github.com/deckhouse/deckhouse/issues/5254?utm=ford_plain-number-sign

## What is the expected result?

New tool with an argument to specify the template to be rendered. 

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: tool
type: feature
summary: Easily expandable tool that should help render helm templates.
impact_level: default
```

### CODE REVIEW

**Review by yalosev — DISMISSED** [review]

File: tools/helm_generate/runners/alert-image-checks/main.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+limitations under the License.
+*/
+
+package alertimagechecks
+
+import (
+	"flag"
+)
+
+const defaltTemplate = "340-extended-monitoring/monitoring/prometheus-rules/image-availability/image-checks.tpl"
```
Comment (z9r5) [inline tools/helm_generate/runners/alert-image-checks/main.go:23]:
```suggestion
const defaltTemplate = "340-extended-monitoring/monitoring/prometheus-rules/image-availability/image-checks.tpl"
```
typo in the word 'default'

Comment (z9r5) [inline tools/helm_generate/runners/alert-image-checks/main.go:23]: typo in the word 'default'

File: tools/helm_generate/main.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+			return cmd.Run()
+		}
+	}
+
+	return fmt.Errorf("unknown runner name: %s", subcommand)
+}
+
+func main() {
+	if err := run(os.Args[1:]); err != nil {
+		log.Print(err)
```
Comment (z9r5) [inline tools/helm_generate/main.go:62]: Let's exit here with a nonzero exit code?

Comment (z9r5) [reply tools/helm_generate/main.go:62]: And let's output errors to stderr?

File: tools/helm_generate/runners/alert-image-checks/image-checks.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	io.WriteString(os.Stdout, renderContent["extended-monitoring/templates/image-checks"])
+	return nil
+}
+
+func renderHelmTemplate(template string) (map[string]string, error) {
+	deckhouseRoot, err := helper.DeckhouseRoot()
+	if err != nil {
+		return nil, err
+	}
+	renderDirPath, err := helper.NewRenderDir("extended-monitoring")
```
Comment (z9r5) [inline tools/helm_generate/runners/alert-image-checks/image-checks.go:43]: Is the "extended-monitoring" a fake chart name? Maybe it is better to use some "fake" name?  E.g. "fakechart"

File: tools/helm_generate/runners/alert-image-checks/image-checks.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	"github.com/deckhouse/deckhouse/testing/library/helm"
+)
+
+func run(templateName string) error {
+	renderContent, err := renderHelmTemplate(templateName)
+	if err != nil {
+		return err
+	}
+
+	io.WriteString(os.Stdout, renderContent["extended-monitoring/templates/image-checks"])
```
Comment (z9r5) [inline tools/helm_generate/runners/alert-image-checks/image-checks.go:34]: See the comment below about the fake chart.

File: tools/helm_generate/runners/alert-image-checks/main.go (hunk 1 above)
Comment (Dmitrykob) [reply tools/helm_generate/runners/alert-image-checks/main.go:23]: fixed

File: tools/helm_generate/main.go (hunk 2 above)
Comment (Dmitrykob) [reply tools/helm_generate/main.go:62]: Good idea. Added exit code.

Comment (Dmitrykob) [reply tools/helm_generate/main.go:62]: made corrections

File: tools/helm_generate/runners/alert-image-checks/image-checks.go (hunk 3 above)
Comment (Dmitrykob) [reply tools/helm_generate/runners/alert-image-checks/image-checks.go]: made corrections

File: tools/helm_generate/runners/alert-image-checks/image-checks.go (hunk 4 above)
Comment (Dmitrykob) [reply tools/helm_generate/runners/alert-image-checks/image-checks.go]: made corrections

**Review by z9r5 — APPROVED** [review]

**Review by yalosev — APPROVED** [review]

**Pull request merged**